### PR TITLE
Add result logic to ActivityResultImplementation for VoiceInput

### LINKED ISSUE #1958: Add a ActivityResultContract for VoiceInput
suggestion: could add this to a ActivityResultContract implementation that can be reused, as it looks like this would be a common pattern

_Originally posted by @luizgrp in https://github.com/google/horologist/pull/1955#discussion_r1450063131_

### PR DESCRIPTION
**Pull request by oas004**
#### WHAT
Add result logic to ActivityResultImplementation for SamplePromptScreen

#### WHY
resolves #1958 

#### HOW
Adding a an implementation of `ActivityResultContract` and extracting the logic into this.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [ ] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: ai/sample/wear-prompt-app/src/main/java/com/google/android/horologist/ai/sample/wear/prompt/prompt/VoiceContract.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        val res = intent?.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)
+        val enteredPrompt = res?.get(0)
+        return if (!enteredPrompt.isNullOrBlank()) {
+            Result.EnteredPrompt(enteredPrompt)
+        } else {
+            Result.Empty
+        }
+    }
+
+    sealed class Result {
```
Comment (yschimke) [inline ai/sample/wear-prompt-app/src/main/java/com/google/android/horologist/ai/sample/wear/prompt/prompt/VoiceContract.kt:37]: The built in ones seem to use a nullable result value, which avoids needing to handle the different result cases.

https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:activity/activity/src/main/java/androidx/activity/result/contract/ActivityResultContracts.kt

Is that useful here?

No strong view, I'm just trying to understand what people expect? Focus on the happy path?

**Review by luizgrp — APPROVED** [review]
LGTM - thanks!

I've got no strong opinion on the matter that Yuri raised, nice to discuss it nonetheless.

File: ai/sample/wear-prompt-app/src/main/java/com/google/android/horologist/ai/sample/wear/prompt/prompt/VoiceContract.kt (hunk 1 above)
Comment (yschimke) [reply ai/sample/wear-prompt-app/src/main/java/com/google/android/horologist/ai/sample/wear/prompt/prompt/VoiceContract.kt:37]: @oas004 not blocking - just for discussion, I'm happy to land if you think it's right as is.

Comment (oas004) [reply ai/sample/wear-prompt-app/src/main/java/com/google/android/horologist/ai/sample/wear/prompt/prompt/VoiceContract.kt:37]: No strong view from my side as well. I was kinda following `GoogleSignInContract` (https://github.com/google/horologist/blob/625e18a0f5dc63dd1d70f8d257a959ec8389d688/auth/ui/src/main/java/com/google/android/horologist/auth/ui/googlesignin/signin/GoogleSignInScreen.kt#L165) .  If we change change to the nullable as you say, does it make sense to change this one as well then perhaps? :) Wdyt? :)

Comment (yschimke) [reply ai/sample/wear-prompt-app/src/main/java/com/google/android/horologist/ai/sample/wear/prompt/prompt/VoiceContract.kt:37]: That's a good signal. Let's leave as is

**Pull request merged**